hail/src/main/scala/is/hail/expr/ir/agg/CollectAsSetAggregator.scala

```diff
@@ -144,7 +144,7 @@ class AppendOnlySetState(val fb: EmitFunctionBuilder[_], t: PType) extends Point
 class CollectAsSetAggregator(t: PType) extends StagedAggregator {
   type State = AppendOnlySetState
 
-  assert(PType.canonical(t) == t)
+  assert(t.isCanonical)
```

Comment:
I think we're settling on () to avoid treating potentially-expensive empty-argument defs as attributes